### ISSUE TITLE
Inclusion of only selected outputs in switchHandling()

### ISSUE THREAD

**Issue by Xenomorphologist** [BitsAndDroids/FlightConnector-Rust#293]
Hi, I noticed `switchHandling()` in `BitsAndDroidsFlightConnector.cpp` switches for all possible outputs, which takes up a lot of program space. If instead, the user could choose the specific outputs to include, a lot of program space can be saved. In my case, when I use `dataHandling()` the program exceeds the microcontroller's storage capacity. However, when I remove unused switch cases from `switchHandling()`, the program size decreases to be within capacity and that is the only way I can run my program.

**Comment by BitsAndDroids** [comment]
You are right. Currently, the generator is quite basic. My proposed first step is to pass a bundle as input and generate a library based on those settings. This change should slim the library down immensely, mainly for lower memory devices. The same would apply to unused output variables, which could then be excluded. 

As an extra step, some types can be optimized in the library. For instance, I need to cut back on the use of Strings (which need to go anyway) or use fixed-width variables (i.e., the heading could never be > 360, so it could be an int of width 16).

**Comment by Xenomorphologist** [comment]
Passing a bundle as input would definitely be a nice feature. Also, thank you for making this useful software 👍

**Comment by BitsAndDroids** [comment]
As a first step im adding flags that disable certain parts of the code. I.e. #define DISABLE_LIGHTS will cut out all logic that handles light events. Same for AP availability events. Once done I'll update the docs to include these flags.

**Comment by BitsAndDroids** [comment]
@Xenomorphologist would you perhaps be able to share the sketch you're working on? I'm trying to find where saving memory pays off the most and would like to benchmark it with a real life use case. Or as a start could you share which microcontroller you are using?

**Comment by Xenomorphologist** [comment]
I'm using a Pro Micro for a custom autothrottle device. My code still has WIP and I haven't tested it with the simulator yet, but with all the libraries I'm using (SPI/LCD Display and custom-built libraries for switches, etc.), the sketch reaches nearly 100% program storage space and ~57% dynamic memory with most of the switch cases commented out, except for a handful. When not commented out, it requires 113% program space.
